up42/utils.py

```diff
@@ -308,92 +308,47 @@ def any_vector_to_fc(
 
 
 def fc_to_query_geometry(
-    fc: Union[dict, FeatureCollection],
-    geometry_operation: str,
-    squash_multiple_features: str = "union",
+    fc: Union[dict, FeatureCollection], geometry_operation: str
 ) -> Union[List, dict]:
     """
-    From a feature collection (one or multiple polygons) & any geometry_operation,
-    gets a single query geometry for the workflow parameters.
-    Returns either a list of bounds or a geojson Polygon (as dict) depending on geometry_operation.
-    If an input fc with multiple features is provided, it gets squashed to a single
-    output geometry, either by taking the first geometry or the union (footprint) of all geometries,
-    depending on handle_multiple_features.
-
-    Examples (geometry & geometry_operation > always returns a single feature):
-        Single input geometries:
-            - feature & "intersects/contains" > same as input feature
-            - feature & "bbox" > rectangular feature that is the bbox of the input feature
-        Multiple input geometries:
-            - features & "intersects/contains" > feature of first object in fc or union
-                of fc (depending on "handle_multiple_features")
-            - features & "bbox" > rectangular feature of first object in fc or union of
-                fc (depending on "handle_multiple_features")
+    From a feature collection with a single feature, depending on the geometry_operation,
+    returns the feature as a list of bounds coordinates or a geojson Polygon (as dict).
 
     Args:
         fc: feature collection
         geometry_operation: One of "bbox", "intersects", "contains".
-        squash_multiple_features: One of "union" (default, footprint of all features)
-            or "first" (takes the first feature.
 
     Returns:
-
+        The feature as a list of bounds coordinates or a geojson Polygon (as dict)
     """
-    if geometry_operation not in ["bbox", "intersects", "contains"]:
-        raise ValueError(
-            "geometry_operation needs to be one of bbox",
-            "intersects",
-            "contains",
-        )
     try:
         if fc["type"] != "FeatureCollection":
             raise ValueError("Geometry argument only supports Feature Collections!")
     except (KeyError, TypeError) as e:
         raise ValueError("Geometry argument only supports Feature Collections!") from e
 
-    # TODO: Handle multipolygons
-
-    # With the now uniform feature collection, decide to return a feature or list of bounds (bbox).
+    geometry_error = "The provided geometry {}, UP42 only accepts single geometries."
     if len(fc["features"]) == 1:
         f = fc["features"][0]
-        if geometry_operation == "bbox":
-            try:
-                query_geometry = list(f["bbox"])
```

Comment:
I don't love this `f` dictionary... I tend to read it as an f-string at first glance and only later I realise it is actually a dictionary.

Reply:
good point, changed to feature